pythainlp/transliterate/thai2rom.py

```diff
@@ -50,102 +69,276 @@ def _prepare_sequence_in(self, input_text):
                 idxs.append(self._char_to_ix[w])
             else:
                 idxs.append(self._char_to_ix["<UNK>"])
-        idxs.append(self._target_char_to_ix["<end>"])
+        idxs.append(self._char_to_ix["<end>"])
         tensor = torch.tensor(idxs, dtype=torch.long)
         return tensor.to(device)
 
     def romanize(self, input_text):
         """
         :param str input_text: Thai text to be romanized
-        :return: English (more or less) text that spells out how the Thai text should be pronounced.
+        :return: English (more or less) text that spells out how the Thai text
+                 should be pronounced.
         """
-        input_tensor = self._prepare_sequence_in(input_text)
-        input_length = input_tensor.size(0)
-        encoder_outputs, encoder_hidden = self._encoder(input_tensor)
-        decoder_input = torch.tensor(
-            [self._target_char_to_ix["<start>"]], device=device)
-        decoder_hidden = (
-            encoder_hidden[0].reshape(
-                1, 1, self._encoder.hidden_dim), encoder_hidden[1].reshape(
-                1, 1, self._encoder.hidden_dim))
-
-        decoded_seq = []  # output
-
-        for di in range(self._maxlength):
-            decoder_output, decoder_hidden, _ = self._decoder(
-                decoder_input, decoder_hidden, encoder_outputs)
-            topv, topi = decoder_output.data.topk(1)
-            if topi.item() == self._target_char_to_ix["<end>"]:
-                decoded_seq.append('<end>')
-                break
-            else:
-                decoded_seq.append(self._ix_to_target_char[topi.item()])
+        input_tensor = self._prepare_sequence_in(input_text).view(1, -1)
+        input_length = [len(input_text) + 1]
 
-            decoder_input = topi.squeeze().detach()
+        target_tensor_logits = self._network(input_tensor,
+                                             input_length,
+                                             None, 0)
 
-        return "".join(decoded_seq[:-1])
+        try:
+            target_tensor = (
+                torch.argmax(target_tensor_logits.squeeze(1), 1).cpu().numpy()
+            )
+            target_indices = [t for t in target_tensor]
+            target = [self._ix_to_target_char[t] for t in target_tensor]
+        except Exception as e: 
```

Comment:
Should we use a more specific exception?

Reply:
I have changed from try-catch to if statement, to check the output tensor dimension from Seq2Seq model.

Firstly, it is to handle the case when the `torch.argmax` raise an exception.
This due to the following code segment within the method named `forward` of Seq2seq class return an empty tensor 

```
 for di in range(max_len):
     ...
     if inference and decoder_input == end_token:
                return outputs[:di]
    ...
```

if the variable `di` is 0 (first iteration), and the first decoder output
is the end token <END>, it would return outputs[:0] which is an
empty Tensor, torch.Size([0]).